### PR TITLE
Delegate server and API stub

### LINKED ISSUE #140: Do delegate server stub
Probably we need different cabal executable for this.

I guess they will be also started in `docker-compose` as `hydra-nodes` #90 

Tasks:

- [x] Add pure logic FSM-like implementation getting pure `DelegateEvents` as input.
- [ ] Add new executable: it starts a multiple threads which push events into shared queue

### PR DESCRIPTION
**Pull request by uhbif19**
Should include all logic from spec.

Closes #140

### CODE REVIEW

File: src/HydraAuction/Hydra/Interface.hs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module HydraAuction.Hydra.Interface (HydraEvent (..)) where
```
Comment (GeorgeFlerovsky) [inline src/HydraAuction/Hydra/Interface.hs:1]:
```suggestion
module HydraAuction.API.Hydra (HydraEvent (..)) where
```

**Review by GeorgeFlerovsky — APPROVED** [review]
Looks reasonable to me.

I've suggested a small module re-org (`X.Interface` --> `API.X`). I think it's nicer to have all the API modules in the same subfolder.

Also, you can take a look at Hydra's API folder to see if you want to borrow some of their ideas for our API.

https://github.com/input-output-hk/hydra/tree/master/hydra-node/src/Hydra/API

File: src/HydraAuction/Delegate/Interface.hs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+module HydraAuction.Delegate.Interface (
```
[the remark this answers is not in the document]
Comment (uhbif19) [reply src/HydraAuction/Delegate/Interface.hs:1]: >  I think it's nicer to have all the API modules in the same subfolder.

Why it is?

I think we would have `Delegate.*` module either way, so API could be placed where.

[the remark this answers is not in the document]
Comment (uhbif19) [reply src/HydraAuction/Delegate/Interface.hs:1]: @GeorgeFlerovsky .

**Review by nazrhom — APPROVED** [review]

**Pull request closed without merging**